doc/proposals/images-subspec.md

```diff
@@ -0,0 +1,190 @@
+---
+title: Images Sub-specification
+authors:
+  - '@zachncst'
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2019-11-04
+last-updated: 2019-11-05
+status: provisional
+see-also:
+  - 'https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/contributors/design-proposals/related-images.md'
+---
+
+# Images Sub Spec
+
+## Release Signoff Checklist
+
+- \[ \] Enhancement is `implementable`
+- \[ \] Design details are appropriately documented from clear requirements
+- \[ \] Test plan is defined
+- \[ \] Graduation criteria for dev preview, tech preview, GA
+- \[ \] User-facing documentation is created in [operator-sdk/doc][operator-sdk-doc]
+
+## Summary
+
+When an operator is installed, the images that the operator installs as operands in the
+form of pods, deployments or other kubernetes resources is not always clearly
+defined. This proposal would require that images would be exposed and
+overridable on the operator custom resource definitions (CRDs). This allows a user of a deployed
+operator to override the images needed at deployment of the custom resource
+(CR).
+
+## Motivation
+
+This proposal is driven by a few motivations:
+
+- Makes operators more transparent with what is being installed on the cluster.
+- Formalizes a best practice of operator development.
+- The depencency between operator version and operator resource version is broken.
+- Allows for out of band updates of the underlying images.
+- Registry information for the images can easily be overridden to point at
+  private registries.
+- Provides some standardization to the operator CRD pattern.
+
+## Goals
+
+- Provide an easy mechanism of defining and overriding images and related image
+  information (like pull secrets).
+- Ensure the operators respect overridden images.
+- Increase transparency with what operators are installing.
+
+### Non-Goals
+
+- OLM management of the images spec is not in this proposal.
+- Mapping/re-mapping other container values (ports, etc.).
+
+## Proposal
+
+Implementing the proposal would include:
+
+- Add an optional section to the CRD spec named images that contains definitions of
+  all images used by the operator. An example:
```

Comment:
I am not sure if I understood properly this proposal. Note that the resource/object which represents the operator is the operator.yaml and not the CRD's which will be created to represents the pieces/units that are required for all solution which is managed by the operator. 

So, in this way, I cannot see why we would add in a CRD all images used for one operator project. What CRD should be used? Note that each operator could have N CRD's which is up to design/business requirements. 
Also, I'd like to recommend check https://github.com/operator-framework/operator-sdk/issues/2151 and https://github.com/operator-framework/operator-sdk/issues/2132 in order to provide a better explanation over it.